server/src/test/java/org/opensearch/search/aggregations/bucket/histogram/DateHistogramAggregatorTests.java

```diff
@@ -1562,25 +1624,48 @@ private void testFilterRewriteCase(
         Consumer<DateHistogramAggregationBuilder> configure,
         Consumer<InternalDateHistogram> verify,
         boolean useDocCountField,
-        Consumer<Integer> verifyCollectCount
+        Consumer<Integer> verifyCollectCount,
+        boolean randomWrite
     ) throws IOException {
         DateFieldMapper.DateFieldType fieldType = aggregableDateFieldType(false, true);
 
         try (Directory directory = newDirectory()) {
-            try (RandomIndexWriter indexWriter = new RandomIndexWriter(random(), directory)) {
-                Document document = new Document();
-                if (useDocCountField) {
-                    // add the doc count field to the first document
-                    document.add(new NumericDocValuesField(DocCountFieldMapper.NAME, 5));
+            if (randomWrite) {
+                try (RandomIndexWriter indexWriter = new RandomIndexWriter(random(), directory)) {
+                    Document document = new Document();
+                    if (useDocCountField) {
+                        // add the doc count field to the first document
+                        document.add(new NumericDocValuesField(DocCountFieldMapper.NAME, 5));
+                    }
+                    for (Map.Entry<String, Integer> date : dataset.entrySet()) {
+                        for (int i = 0; i < date.getValue(); i++) {
+                            long instant = asLong(date.getKey(), fieldType);
+                            document.add(new SortedNumericDocValuesField(AGGREGABLE_DATE, instant));
+                            document.add(new LongPoint(AGGREGABLE_DATE, instant));
```

Comment:
Not important for this PR, but just as a bit of trivia, these two lines (and the equivalent in the `else` block) can be combined into one:

```
document.add(new LongField(AGGREGABLE_DATE, instant, Store.NO);
```